server/pkg/schema/corresponding_field_test.go

```diff
@@ -0,0 +1,103 @@
+package schema
+
+import (
+	"testing"
+
+	"github.com/reearth/reearth-cms/server/pkg/id"
+	"github.com/reearth/reearth-cms/server/pkg/key"
+	"github.com/reearth/reearth-cms/server/pkg/project"
+	"github.com/reearth/reearthx/account/accountdomain"
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetCorrespondingFields(t *testing.T) {
+	prj := project.New().NewID().MustBuild()
+	mid1 := id.NewModelID()
+	mid2 := id.NewModelID()
+	fid1 := id.NewFieldID()
+	cf1 := &CorrespondingField{
+		Title:       lo.ToPtr("title"),
+		Key:         lo.ToPtr("key"),
+		Description: lo.ToPtr("description"),
+		Required:    lo.ToPtr(true),
+	}
+	f1 := NewField(NewReference(mid2, cf1, cf1.FieldID).TypeProperty()).ID(fid1).Key(key.Random()).MustBuild()
+	s1 := New().NewID().Workspace(accountdomain.NewWorkspaceID()).Project(prj.ID()).Fields(FieldList{f1}).MustBuild()
+	s2 := New().NewID().Workspace(accountdomain.NewWorkspaceID()).Project(prj.ID()).Fields(FieldList{}).MustBuild()
+	fr1, _ := FieldReferenceFromTypeProperty(f1.TypeProperty())
+	fields, _ := GetCorrespondingFields(s1, s2, mid1, f1, fr1)
+
+	// check that fields are not nil
+	assert.NotNil(t, fields.Schema1)
+	assert.NotNil(t, fields.Schema2)
+	assert.NotNil(t, fields.Field1)
+	assert.NotNil(t, fields.Field2)
+
+	// check that model ids are correct
+	assert.Equal(t, fields.Field2.typeProperty.reference.modelId, mid1)
+	assert.Equal(t, fields.Field1.typeProperty.reference.modelId, mid2)
```

Comment:
You may want to minimize the use of private fields in your test code that are not directly related to the implementation you are trying to test.You never know when a private field will change and its specification is implicit. Using public methods makes it clear that the intent is not to test them.

```suggestion
	rf, _ := FieldReferenceFromTypeProperty(fields.Field2.TypeProperty())
	assert.Equal(t, rf.modelId, mid1)
	rf, _ := FieldReferenceFromTypeProperty(fields.Field1.TypeProperty())
	assert.Equal(t, rf.modelId, mid2)
```